modules/highgui/test/test_gui.cpp

```diff
@@ -253,6 +253,77 @@ TEST(Highgui_GUI, currentUIFramework)
     EXPECT_GT(framework.size(), 0);  // builtin backends
 #endif
 }
-
-
+#if (!defined(ENABLE_PLUGINS) \
+        && !defined HAVE_GTK \
+        && !defined HAVE_QT \
+        && !defined HAVE_WIN32UI \
+        && !defined HAVE_COCOA \
+        && !defined HAVE_WAYLAND \
+    ) || !defined(HAVE_OPENGL)
+TEST(Highgui_GUI, DISABLED_gl)
+#else
+#include<epoxy/gl.h>
+#include<GL/gl.h>
+static GLuint create_shader(const char* source, GLenum type) {
+    GLuint shader = glCreateShader(type);
+    glShaderSource(shader, 1, &source, NULL);
+    glCompileShader(shader);
+    return shader;
+}
+struct DrawData
+{
+    GLuint vao, vbo, program;
+};
+void draw(void* userdata){
+    DrawData* data = static_cast<DrawData*>(userdata);
+    glBindVertexArray(data->vao);
+    glDrawArrays(GL_TRIANGLES, 0, 3);
+    glBindVertexArray(0);
+}
+TEST(Highgui_GUI, gl)
+#endif
+{
+    const std::string window_name("gl_test_window");
+    const Size image_size(800, 600);
+    EXPECT_NO_THROW(destroyAllWindows());
+    ASSERT_NO_THROW(namedWindow(window_name, WINDOW_OPENGL));
+    ASSERT_NO_THROW(resizeWindow(window_name, image_size));
+    DrawData data;
+    const char *vertex_shader_source =
+            "#version 330 core\n"
+            "layout (location = 0) in vec3 position;\n"
+            "void main() {\n"
+            "   gl_Position = vec4(position, 1.0);\n"
+            "}\n";
+    const char *fragment_shader_source =
+            "#version 330 core\n"
+            "out vec4 color;\n"
+            "void main() {\n"
+            "   color = vec4(1.0, 1.0, 1.0, 1.0); // white\n"
+            "}\n";
+    GLuint vertex_shader = create_shader(vertex_shader_source, GL_VERTEX_SHADER);
+    GLuint fragment_shader = create_shader(fragment_shader_source, GL_FRAGMENT_SHADER);
+    data.program = glCreateProgram();
+    glAttachShader(data.program, vertex_shader);
+    glAttachShader(data.program, fragment_shader);
+    glLinkProgram(data.program);
+    glUseProgram(data.program);
+    GLfloat vertices[] = {
+            0.0f,  0.5f, 0.0f,
+            -0.5f, -0.5f, 0.0f,
+            0.5f, -0.5f, 0.0f
+    };
+    glGenVertexArrays(1, &data.vao);
+    glBindVertexArray(data.vao);
+    glGenBuffers(1, &data.vbo);
+    glBindBuffer(GL_ARRAY_BUFFER, data.vbo);
+    glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_STATIC_DRAW);
+    glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(GLfloat), (GLvoid*)0);
+    glEnableVertexAttribArray(0);
+    glBindVertexArray(0);
+    ASSERT_NO_THROW(setOpenGlDrawCallback(window_name, draw, &data));
+    EXPECT_NO_THROW(waitKey(10000));
+    EXPECT_NO_THROW(setOpenGlDrawCallback(window_name, 0));
+    EXPECT_NO_THROW(destroyAllWindows());
```

Comment:
Please remove the new test. We cannot run it in CI.

Reply:
https://github.com/opencv/opencv/pull/25822/commits/b9776440c889212cc41f3b0b292c4b48d5c53c4a